m4/check_pkg_cuda.m4

```diff
@@ -21,40 +21,52 @@ AC_DEFUN([CHECK_PKG_CUDA], [
         [],
         [test "${with_cuda}" = "no"],
         [check_pkg_found=no],
-        [AS_IF([test -d ${with_cuda}/lib64], [check_pkg_libdir="lib64"], [check_pkg_libdir="lib"])
+        [AS_IF([test -d $(realpath ${with_cuda})/lib64], [check_pkg_libdir="lib64"], [check_pkg_libdir="lib"])
+         CUDA_LDFLAGS="-L$(realpath ${with_cuda})/${check_pkg_libdir}"
+         CUDA_LIBS="-lcudart_static -lrt -ldl"
+         CUDA_CPPFLAGS="-fPIC -I$(realpath $(realpath ${with_cuda})/include)"
+         LDFLAGS="${CUDA_LDFLAGS} ${LDFLAGS}"
+         LIBS="${CUDA_LIBS} ${LIBS}"
+         CPPFLAGS="${CUDA_CPPFLAGS} ${CPPFLAGS}"
+        ])
 
-         CUDA_LDFLAGS="-L${with_cuda}/${check_pkg_libdir}"
-
-         CPPFLAGS="-I${with_cuda}/include ${CPPFLAGS}"
-         LDFLAGS="${CUDA_LDFLAGS} ${LDFLAGS}"])
+  AS_IF([test "${check_pkg_found}" = "yes"],
+        [AC_SEARCH_LIBS(
+         [cudaGetDriverEntryPoint],
+         [cudart_static],
+         [],
+         [check_pkg_found=no],
+         [-ldl -lrt])])
 
   AS_IF([test "${check_pkg_found}" = "yes"],
-        [AC_CHECK_HEADERS([cuda.h], [], [check_pkg_found=no])])
+        [
+        AC_MSG_CHECKING([if CUDA 11.7+ is available])
```

Comment:
This is a new requirement.  How did we get here?

Reply:
this is the minimum CTK to support dmabuf: https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-operator-rdma.html#common-prerequisites

Reply:
Ok, but not all our platforms require dma buf support.  Similar to the libfabric discussion, we shouldn't bump the required version of CUDA just for dma buf.